batchflow/named_expr.py

```diff
@@ -568,3 +568,61 @@ def get(self, batch=None, pipeline=None, model=None, parallel=False):   # pylint
         if parallel:
             return self.name.get(batch=batch, pipeline=pipeline, model=model)
         return self
+
+class I(NamedExpression):
+    """ Iteration number
+
+    Parameters
+    ----------
+    name : str
+        'c', 'current' - current iteration number, default.
+        'm', 'max', maximum' - total number of iterations to be performed.
+                               If total number is not defined, raises an error.
+        'r', 'ratio' - current iteration divided by a total number of iterations.
+
+    Raises
+    ------
+    ValueError
+        * If `name` is not valid.
+        * If `name` is 'm' or 'r' and total number of iterations is not defined.
+    Examples
+    --------
+    ::
+
+        I()
+        I('m')
+        P(R('normal', loc=0, scale=I('ratio')*100)
+    """
+    def __init__(self, name='c'):
+        super().__init__(name, mode=None)
+        self._allowed_names = {
+            'current': 'c',
+            'c': 'c',
+            'maximum': 'm',
+            'm': 'm',
+            'max': 'm',
+            'ratio': 'ratio',
+            'r': 'r',
+        }
+
+    def get(self, batch=None, pipeline=None, model=None):
+        """ Return current or maximum iteration number or their ratio """
+        name = self._allowed_names.get(self._get_name())
```

Comment:
```suggestion
        name = self._allowed_names.get(self._get_name(batch, pipeline, model))
```

Reply:
Is there a way to accept suggestion without  starting a shippable run?

Reply:
I don't think so